apps/agent/app/src/components/Transactions.js

```diff
@@ -0,0 +1,479 @@
+                onClick={onDownload}
+              />
+            </div>
+          </div>
+        </React.Fragment>
+      }
+      onPageChange={setPage}
+      onStatusEmptyClear={handleClearFilters}
+      page={page}
+      status={emptyResultsViaFilters ? 'empty-filters' : 'default'}
+      renderEntry={({
+        date,
+        description: reference,
+        type,
+        tokenTransfers,
+        onlyOne,
+        isIncoming,
+        targetContract,
+      }) => {
+        const [{ token, amount, to, from } = {}] = tokenTransfers
+        const entity = to || from
+        const formattedDate = format(date, "yyyy-MM-dd'T'HH:mm:ss")
+
+        const dateDiv = (
+          <time
+            dateTime={formattedDate}
+            title={formattedDate}
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.surfaceContent};
+            `}
+          >
+            {format(date, 'MM/dd/yy')}
+          </time>
+        )
+        const badgeDiv = onlyOne ? (
+          <div
+            css={`
+              ${layoutName === 'medium' &&
+                `
+                  display: inline-flex;
+                  max-width: 150px;
+                `}
+              ${layoutName === 'large' &&
+                `
+                  display: inline-flex;
+                  max-width: 'unset';
+                `}
+            `}
+          >
+            <LocalIdentityBadge
+              connectedAccount={addressesEqual(entity, connectedAccount)}
+              networkType={network && network.type}
+              entity={entity}
+            />
+          </div>
+        ) : (
+          <LocalIdentityBadge
+            badgeOnly={typeof targetContract !== 'string'}
+            entity={
+              (!tokenTransfers.length && targetContract) || 'Multiple accounts'
+            }
+            networkType={
+              typeof targetContract === 'string' && network ? network.type : ''
+            }
+          />
+        )
+        const typeDiv = (
+          <div
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.surfaceContent};
+            `}
+          >
+            {TRANSACTION_TYPES_LABELS[type]}
+          </div>
+        )
+        const referenceDiv = (
+          <div
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.surfaceContent};
+            `}
+          >
+            {reference}
+          </div>
+        )
+        const amountDiv = (() => {
+          if (!onlyOne) {
+            const uniqueTokens = Object.keys(
+              tokenTransfers.reduce((p, { token }) => {
+                p[token] = 1
+                return p
+              }, {})
+            ).length
+            return (
+              <div
+                css={`
+                  ${textStyle('body2')};
+                  color: ${theme.surfaceContent};
+                `}
+              >
+                {uniqueTokens} token
```

Comment:
We should add a non-breaking space here:

```suggestion
                {uniqueTokens} token
```